app/src/test/java/com/example/mytestproject/ui/searchCity/SearchCityViewModelTest.kt

```diff
@@ -0,0 +1,214 @@
+
+        // Given
+        val list = mutableListOf<List<CityModel>>()
+        val observer = Observer<List<CityModel>> {
+            list.add(it)
+        }
+
+        try {
+
+            // When
+            searchCityViewModel.lastChosenCities.observeForever(observer)
+
+            val value = searchCityViewModel.lastChosenCities.value
+
+            // Then
+            val lastChosenCitiesList = listOf(
+                CityModel(6662, "Sacramento", "USA"),
+                CityModel(345, "Milan", "Italy"),
+                CityModel(234, "Moscow", "Russia")
+            )
+
+            assertEquals(lastChosenCitiesList[0], value?.get(0))
+            assertEquals(lastChosenCitiesList[1], value?.get(1))
+            assertEquals(lastChosenCitiesList[2], value?.get(2))
+        } finally {
+            searchCityViewModel.lastChosenCities.removeObserver(observer)
+        }
+    }
+
+
+    @Test
+    fun `search city is not empty`() {
+        // Given
+        val cityName = "Moscow"
+        val cityId = 234
+        `when`(cityFilter.filterCityList(cityName)).thenReturn(
+            listOf(
+                CityModel(cityId, cityName, "Russia")
+            )
+        )
+
+        val list = mutableListOf<List<CityModel>>()
+        val observer = Observer<List<CityModel>> {
+            list.add(it)
+        }
+
+        try {
+
+            searchCityViewModel.lastChosenCities.observeForever(observer)
+
+            // When
+            searchCityViewModel.searchCity(cityName)
+
+            val value = searchCityViewModel.filteredCityList.value
+
+            // Then
+            val filteredCityList = listOf(CityModel(cityId, cityName, "Russia"))
+
+            assertEquals(filteredCityList[0], value?.get(0))
+
+        } finally {
+            searchCityViewModel.filteredCityList.removeObserver(observer)
+        }
+
+    }
+
+    @Test
+    fun `on city chose check navigateToCurrentWeather value`() {
+
+        // Given
+        val cityId = 234
+
+        val listCityModel = mutableListOf<List<CityModel>>()
+        val observerListCityModel = Observer<List<CityModel>> {
+            listCityModel.add(it)
+        }
+
+        val listEvent = mutableListOf<Event<Int>>()
+        val observerEvent = Observer<Event<Int>> {
+            listEvent.add(it)
+        }
+
+        try {
+
+            searchCityViewModel.navigateToCurrentWeather.observeForever(observerEvent)
+
+            searchCityViewModel.filteredCityList.observeForever(observerListCityModel)
+
+            val cityList = searchCityViewModel.filteredCityList.value
+            `when`(insertCityToLastChosenCitiesEntityUseCase.invoke(cityId, cityList)).thenReturn(
+                Completable.never()
+            )
+
+            // When
+            searchCityViewModel.onCityChosen(cityId)
+            val value = searchCityViewModel.navigateToCurrentWeather.value
+
+            // Then
+            assertNotNull(value)
```

Comment:
NotNulll checking doesn't make sense if later we check that value is Event. 
Insetad of these 2 checks let's check that value equals Event(cityId)

Reply:
Fixed